### DISCUSSION TITLE
Zksync Foundry build access and exec errors

### DISCUSSION THREAD

**Discussion by imajdalany** [Cyfrin/foundry-full-course-cu#2092] [Q&A] [answered]
Hello,
I am having a problem when building any project using `forge build --zksync`, but `forge build` without `--zksync` works perfectly.
**Using Linux Debian 12 as OS.**

- when runnig `forge build --zksync` i get the following error
  ```
  [⠊] Compiling (zksync)...
  Error: 
  "/home/ibrahim/.zksync/zksolc-linux-amd64-musl-v1.4.1": Permission denied (os error 13)
  ```
- then when I added `forge` to the sudoers and ran `sudo forge build --zksync` I got this error
  ```
  [⠊] Compiling (zksync)...
  Error: 
  "/root/.zksync/zksolc-linux-amd64-musl-v1.4.1": Exec format error (os error 8)
  ```

Tried reinstalling Foundry and reinstalling Foundry Zksync, it didnt work.

Thank you everyone <3

**Comment by EngrPips** [comment]
I am unsure what is wrong here, but let's look forward to @PatrickAlphaC  or @cromewar suggestions when they are available.

**Comment by imajdalany** [comment, marked as answer]
After reinstalling multiple times it worked and I think that the problem is:
- The first time running `forge build --zksync` it took long time without any output, I thought it was caused by my slow internet, I interrupted the program using `Control+C` and maybe **that broke zksync**
- After reinstalling everything and waiting the first time running `forge build --zksync` it worked

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it buddy

**Reply by sg-milad** [reply]
didn't work for me

**Reply by EngrPips** [reply]
> didn't work for me

ZKSync stuff is optional. Please don't sweat over it too much.

**Comment by giorgiabes** [comment]
$ ls ~/.zksync/
solc-linux-amd64-0.8.19-1.0.1  zksolc-linux-amd64-musl-v1.4.1

$ rm -rf ~/.zksync/

$ forge build --zksync
